opsdroid/connector/matrix/connector.py

```diff
@@ -414,45 +458,57 @@ async def _send_room_creation(self, creation_event):
     @register_event(events.RoomName)
     @ensure_room_id_and_send
     async def _send_room_name_set(self, name_event):
-        return await self.connection.set_room_name(name_event.target, name_event.name)
+        return await self.connection.room_put_state(
```

Comment:
I am taking this as a generic example of a pattern here.

Previously before nio if a call like this errored, the error would raise and be caught in the logs as the outgoing send would fail. I am not sure what the best solution to this is, if all the `nio.XXXError` classes have a common parent we could write a decorator which catches them in the return and logs them at error?

(We can probably punt this to an issue post merge)